### ISSUE TITLE
Load data in another datatype rather than float64

### ISSUE THREAD

**Issue by akasom89** [pytroll/satpy#2771]
Based on my experiences with Satpy, all satellite data I have loaded so far are in the float64 datatype. However, I'm considering whether it's possible to set the datatype to a different type, such as np.float16, as float64 may be unnecessary for many use cases. 

I searched the documentation. Although there mentioned the same point, but I could not find how to do that. [link to docs](https://satpy.readthedocs.io/en/stable/dev_guide/custom_reader.html)

> At the moment, satellite instruments are rarely measuring in a resolution greater than what can be encoded in 16 bits. As such, to preserve processing power, please consider carefully what data type you should scale or calibrate your data to.

**Comment by mraspaud** [comment]
@akasom89 thanks for you question, it is indeed very relevant.

Since a few months, we have started migrating most of the satellite data from float64 to float32, since we realized, as you say, that float64 is unnecessary for this data. This results in more memory efficient computations.

However, we have not introduced the possibility to choose the type. As you note, most satellites measure the earth radiance and store the measurement in a 16 bits (unsigned) integer. In order to preserver those 16 bits when converting to floats (for calibration), we need at least a float32, with a mantissa of 22 bits, as the mantissa of the float16 in only 10 bit.
So we are now taking measures to convert all satpy processing to support (and stay in) 32 bits floats.

**Comment by mraspaud** [comment]
Closing this for now, feel free to reopen if more action is needed here.

**Comment by mraspaud** [comment]
Most readers should already be converted to float32 in the current stable version, so I'm wondering what data you are reading with satpy and version you have installed?

**Comment by akasom89** [comment]
Yes. You are right. It seems that it was a problem of the outdated version on my side. Thanks.
